mathesar_ui/src/stores/table-data/record-summaries/recordSummaryUtils.ts

```diff
@@ -154,3 +163,14 @@ export function renderRecordSummaryForRow({
   const inputData = prepareFieldsAsRecordSummaryInputData(fields);
   return renderTransitiveRecordSummary({ template, inputData, transitiveData });
 }
+
+const plainTextNull = 'NULL';
+const stylizedNull = '<span class="postgres-keyword">NULL</span>';
+
+export function displayRecordSummaryAsPlainText(str: string): string {
+  return str.replace(nullPlaceholderPattern, plainTextNull);
+}
+
+export function displayRecordSummaryAsHtml(str: string): string {
+  return escapeHtml(str).replace(nullPlaceholderPattern, stylizedNull);
+}
```

Comment:
Is there any specific reason why we need this method to return a string?

Instead can we return a `(string | ComponentAndProps)[]`, and split the input string, and replace the null value with a component?

This will remove our need for using the `@html` tag.

Reply:
Ah, okay, this is brilliant. Using an _array_ of `string | ComponentAndProps` is the missing piece that I was not thinking of. Thanks.

I'd actually like to take this approach a bit further with some deeper refactoring though, because it will improve performance be avoiding the need to re-parse the record summary string. I've opened #2037 to track that work. Can we merge this PR for now, and then use the approach in that ticket later?